### PR TITLE
Add group profile for transactional DB workload

### LINKED ISSUE #1250: group profile for transactional db workload
group profile infrastructure already exists in GD2, however we'd need to have a profile for db workload tunings to be included for GD2 which should be as follows:

performance.open-behind=on
performance.write-behind=off
performance.stat-prefetch=off
performance.quick-read=off
performance.strict-o-direct=on
performance.read-ahead=off
performance.io-cache=off
performance.readdir-ahead=off
performance.client-io-threads=on
server.event-threads=4
client.event-threads=4
performance.read-after-open=yes

### Observed behavior
No db profile is currently present

### Expected/desired behavior
DB profile should be present

### Information about the environment:

- Glusterd2 version used (e.g. v4.1.0 or master): 
- Operating system used: 
- Glusterd2 compiled from sources, as a package (rpm/deb), or container: 
- Using External ETCD: (yes/no, if yes ETCD version):
- If container, which container image: 
- Using kubernetes, openshift, or direct install: 
- If kubernetes/openshift, is gluster running inside kubernetes/openshift or outside: 

### Other useful information

- glusterd2 config files from all nodes (default /etc/glusterd2/glusterd2.toml)
- glusterd2 log files from all nodes (default /var/log/glusterd2/glusterd2.log)
- ETCD configuration
- Contents of `uuid.toml` from all nodes (default /var/lib/glusterd2/uuid.toml)
- Output of `statedump` from any one of the node

### Useful commands

- To get glusterd2 version
    ```
    glusterd2 --version
    ```
- To get ETCD version
    ```
    etcd --version
    ```
- To get output of statedump
    ```
    curl http://glusterd2-IP:glusterd2-Port/statedump
    ```

### PR DESCRIPTION
**Pull request by prashanthpai**
For this to work, it needs support from volgen for providing a way to
turn an xlator on or off by its name.

Fixes #1250 
Updates #376 
Signed-off-by: Prashanth Pai <ppai@redhat.com>

### CODE REVIEW

File: glusterd2/commands/volumes/grouped-options.go [hunk 1]
```diff
@@ -129,6 +129,28 @@ var defaultGroupOptions = map[string]*api.OptionGroup{
...
+			{Name: "quick-read", OnValue: "off"},
+			{Name: "read-ahead", OnValue: "off"},
+			{Name: "io-cache", OnValue: "off"},
+			{Name: "readdir-ahead", OnValue: "off"},
+			// as per new volgen, this has been named 'md-cache' instead of its
+			// legacy name 'stat-prefetch'
+			{Name: "md-cache", OnValue: "off"},
+			// FIXME: how should this option be named ? fuse.io-threads ?
+			// https://github.com/gluster/glusterd2/issues/376#issuecomment-424970778
+			{Name: "client-io-threads", OnValue: "on"},
```
Comment (prashanthpai) [inline glusterd2/commands/volumes/grouped-options.go]: Do not merge this until this is finalised.

Comment (prashanthpai) [reply glusterd2/commands/volumes/grouped-options.go]: @aravindavk Thoughts on this ?

Comment (aravindavk) [reply glusterd2/commands/volumes/grouped-options.go]: once template based volgen PR merges, xlator can be enabled/disabled like this. But only if the xlator name exists in the template.

Comment (prashanthpai) [reply glusterd2/commands/volumes/grouped-options.go]: In this specific case, io-threads can be loaded on the brick (default) and also on the client (additional and optional). If a user has to turn on io-threads on the client, what will the option look like ?

**Comment by Madhu-1** [comment]
@aravindavk can we take this once we get `volgen` patch in?

File: glusterd2/commands/volumes/grouped-options.go (hunk 1 above)
Comment (prashanthpai) [reply glusterd2/commands/volumes/grouped-options.go]: @aravindavk Can you please clarify ?

**Comment by aravindavk** [comment]
Sent glusterfs patch to include a option in xlator option table to enable/disable xlators. https://review.gluster.org/21813

This PR also needs changes to some of the names used here, I will submit new PR to address those issues.

**Comment by aravindavk** [comment]
PR sent to address the option names and fix a few validation issues related to enable/disable xlators. https://github.com/gluster/glusterd2/pull/1370

**Pull request closed without merging**